jenkins_pipelines/environments/common/pipeline-reference.groovy

```diff
@@ -0,0 +1,44 @@
+def run(params) {
+    timestamps {
+        deployed = false
+        env.resultdir = "${WORKSPACE}/results"
+        env.resultdirbuild = "${resultdir}/${BUILD_NUMBER}"
+        env.common_params = "--outputdir ${resultdir} --tf ${params.tf_file} --gitfolder ${resultdir}/sumaform"
+        try {
+            stage('Clone sumaform and susemanager-ci') {
+                // Create a directory for all the outputs belonging to this build number
+                sh "mkdir -p ${resultdir}"
+                git url: params.terracumber_gitrepo, branch: params.terracumber_ref
+                dir("susemanager-ci") {
+                    checkout scm
+                }
+                // Create a directory for  to place the directory with the build results (if it does not exist)
+                sh "mkdir -p ${resultdir}"
+                // Clone sumaform
+                sh "set +x; source /home/jenkins/.credentials set -x; ./terracumber-cli ${common_params} --gitrepo ${params.sumaform_gitrepo} --gitref ${params.sumaform_ref} --runstep gitsync"
+            }
+            stage('Deploy') {
+                // Provision the environment
+                if (params.terraform_bin_init) {
+                    env.TERRAFORM_INIT = '--init'
+                } else {
+                    env.TERRAFORM_INIT = ''
+                }
+                sh "set +x; source /home/jenkins/.credentials set -x; TERRAFORM=${params.terraform_bin} TERRAFORM_PLUGINS=${params.terraform_bin_plugins} ./terracumber-cli ${common_params} --logfile ${resultdirbuild}/sumaform.log ${env.TERRAFORM_INIT} --taint '.*(domain|main_disk).*' --runstep provision"
```

Comment:
It might be good to validate that those Jenkins parameters exist. I guess we expect them in all our Jenkins job, but maybe checking the existence at the beginning can avoid some issues. Fine if we don't do anyway :)

Reply:
I think we better have default values. What do you think?

Reply:
Yep, but in that case, you don't have control of default values, as they are part of a different file, that's what I thought could be good to verify, but let's close the discussion, let's assume we don't fail :D

Reply:
Well, let's put it this way: the other file defines default values in case you want to have such parameters, for the builds.

But it could also be that most jobs would want to avoid having TERRAFORM and TERRAFORM_PLUGINS as well because they never ever change them.

That's why I think that in that case, we should define default values here.